six/modules/c++/six/source/NITFReadControl.cpp

```diff
@@ -195,33 +195,42 @@ void NITFReadControl::load(nitf::IOInterface& ioInterface,
         {
             if (desid == "XML_DATA_CONTENT")
             {
-                // We better have a user-defined subheader.  For now we
-                // don't actually store off the values, but we do check its
-                // tag and size.  Note that we need to check the subheader
+                // Check whether DES is SICD/SIDD, or something of a different
+                // sort. Note that we need to check the subheader
                 // length first rather than calling tre.getCurrentSize() in
                 // case there is no subheader (in which case
                 // subheader.getSubheaderFields() will throw with a
                 // NITRO-specific message).
                 const sys::Uint64_T
                         subheaderLength(subheader.getSubheaderFieldsLength());
                 if (subheaderLength !=
-                            Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
+                    Constants::DES_USER_DEFINED_SUBHEADER_LENGTH)
                 {
-                    std::ostringstream ostr;
-                    ostr << "Expected a user-defined subheader size of "
-                         << Constants::DES_USER_DEFINED_SUBHEADER_LENGTH
-                         << " but got " << subheaderLength;
-                    throw except::Exception(Ctxt(ostr.str()));
+                    continue;
                 }
 
                 nitf::TRE tre = subheader.getSubheaderFields();
+                std::string field;
+                try
```

Comment:
Rather than a try/catch, use `tre.exists()` to check first.  Since it's going to be really rare that you see an `XML_DATA_CONTENT` without a DESSHSI, and it's not in an inner loop anyway, this isn't a performance thing but a style thing... basically that we want to reserve this for more exceptional situations so that when we do see a try/catch, we think there must be something really weird going on and look at it carefully.